src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs

```diff
@@ -11453,5 +11614,494 @@ static void Main()
+
+        [Fact]
+        public void ConversionsOverload05()
+        {
+            var source = @"
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+class Program
+{
+    static void Main(string[] args)
+    {
+        Action<IEnumerable> x = null;
+        Action<IEnumerable<int>> x1 = null;
+
+        // valid and T is IEnumerable<int> 
+        // Action<in T> is introducing upper bound on T 
+        Test(x, x1);
+
+        (Action<IEnumerable> x, Action<IEnumerable> y) z = (null, null);
+        (Action<IEnumerable<int>> x, Action<IEnumerable<int>> y) z1 = (null, null);
+
+        // inference fails
+        // (T, T) is introducing exact bound on T even in an 'in' param position
+        Test(z, z1);
+
+        // evdently, this is an error. 
+        Test<(IEnumerable<int>, IEnumerable <int>)> (z, z1);
+    }
+
+    public static void Test<T>(Action<T> x, Action<T> y)
+    {
+        System.Console.WriteLine(typeof(T));
+    }
+}
+
+" + trivial2uple;
+
+            var comp = CreateCompilationWithMscorlib(source, parseOptions: TestOptions.Regular.WithTuplesFeature());
+            comp.VerifyDiagnostics(
+                // (22,9): error CS0411: The type arguments for method 'Program.Test<T>(Action<T>, Action<T>)' cannot be inferred from the usage. Try specifying the type arguments explicitly.
+                //         Test(z, z1);
+                Diagnostic(ErrorCode.ERR_CantInferMethTypeArgs, "Test").WithArguments("Program.Test<T>(System.Action<T>, System.Action<T>)").WithLocation(22, 9),
+                // (25,54): error CS1503: Argument 1: cannot convert from '(System.Action<System.Collections.IEnumerable> x, System.Action<System.Collections.IEnumerable> y)' to 'System.Action<(System.Collections.Generic.IEnumerable<int>, System.Collections.Generic.IEnumerable<int>)>'
+                //         Test<(IEnumerable<int>, IEnumerable <int>)> (z, z1);
+                Diagnostic(ErrorCode.ERR_BadArgType, "z").WithArguments("1", "(System.Action<System.Collections.IEnumerable> x, System.Action<System.Collections.IEnumerable> y)", "System.Action<(System.Collections.Generic.IEnumerable<int>, System.Collections.Generic.IEnumerable<int>)>").WithLocation(25, 54),
+                // (25,57): error CS1503: Argument 2: cannot convert from '(System.Action<System.Collections.Generic.IEnumerable<int>> x, System.Action<System.Collections.Generic.IEnumerable<int>> y)' to 'System.Action<(System.Collections.Generic.IEnumerable<int>, System.Collections.Generic.IEnumerable<int>)>'
+                //         Test<(IEnumerable<int>, IEnumerable <int>)> (z, z1);
+                Diagnostic(ErrorCode.ERR_BadArgType, "z1").WithArguments("2", "(System.Action<System.Collections.Generic.IEnumerable<int>> x, System.Action<System.Collections.Generic.IEnumerable<int>> y)", "System.Action<(System.Collections.Generic.IEnumerable<int>, System.Collections.Generic.IEnumerable<int>)>").WithLocation(25, 57)
+                );
+        }
```

Comment:
Would it make sense to test conditionals? 
`(long, long) x = flag ? (1, (long) 2) : ((long) 3, 4);`